### ISSUE TITLE
Minor: Error with Null Pointer?

### ISSUE THREAD

**Issue by SBMous** [kickash32/DistributedMobSpawns#4]
There is a minor bug where the console has a few instances with the following error periodically.
```
[Server thread/ERROR]: Could not pass event CreatureSpawnEvent to DistributedMobSpawns v2.1.1
java.lang.NullPointerException: null
```
The error appears to report in groups of approximately 4 or 5 lasting at most a minute before the console is error free for a while. This repeats after 1-2 minutes. If I were to guess I'd say it looks most like the plugin getting stuck on one mob until that mob despawns or dies.

I am running v2.1.1 of Distributed Mob Spawns

**Server Information**
Running Paper version git-Paper-626 (MC: 1.13.2) (Implementing API version 1.13.2-R0.1-SNAPSHOT)

Java Version '1.8.0_212'

Plugins (27): AnimatedFrames, BuycraftX, CoreProtect, CrazyCrates, CustomCommands, DistributedMobSpawns, Dynmap, Dynmap-WorldGuard, EssentialsX, FishinRoulette, GCore, Harbor, Holographic Displays, LuckPerms, MapManager, NuVotifier, PacketListenerAPI, PlanAnalytics, PlayerHeads, Shopkeepers, SuperbVote, Vault, WorldBorder, WorldEdit, WorldGuard

**Comment by kickash32** [comment]
Sorry for taking so long to get to this, but I needed to find a log which told me what line was causing the issue and I finally got one here https://pastebin.com/p3sa7mR1. It seems like the server was spawning an unexpected mob and the plugin could not handle it so it threw an error and let it spawn regardless. I haven't pinned down what mob it is yet but now it will print a more useful error message in the latest build. So please update to the latest build and paste another log if you see similar errors of the form `"[DMS] Error: unknown mob: "`.

**Comment by robertoccu** [comment]
A lot of NullPointerException on console and this: [DMS] Error: unknown mob: PHANTOM

---

[19-Jul-2019 18:48:56] Could not pass event CreatureSpawnEvent to DistributedMobSpawns v2.1.2
[19-Jul-2019 18:48:56] java.lang.NullPointerException: null
[19-Jul-2019 18:48:56] at me.kickash32.distributedmobspawns.DistributedMobSpawns.getMobCapAmbient(DistributedMobSpawns.java:186) ~[?:?]
[19-Jul-2019 18:48:56] at me.kickash32.distributedmobspawns.EntityProcessor.isSpawnAllowed(EntityProcessor.java:96) ~[?:?]
[19-Jul-2019 18:48:56] at me.kickash32.distributedmobspawns.EventListener.onCreatureSpawnEvent(EventListener.java:22) ~[?:?]
[19-Jul-2019 18:48:56] at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor410.execute(Unknown Source) ~[?:?]
[19-Jul-2019 18:48:56] at org.bukkit.plugin.EventExecutor.lambda$create$1(EventExecutor.java:69) ~[patched_1.14.3.jar:git-Paper-134]
[19-Jul-2019 18:48:56] at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:80) ~[patched_1.14.3.jar:git-Paper-134]
[19-Jul-2019 18:48:56] at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[patched_1.14.3.jar:git-Paper-134]
[19-Jul-2019 18:48:56] at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:545) ~[patched_1.14.3.jar:git-Paper-134]
[19-Jul-2019 18:48:56] at org.bukkit.craftbukkit.v1_14_R1.event.CraftEventFactory.callCreatureSpawnEvent(CraftEventFactory.java:613) ~[patched_1.14.3.jar:git-Paper-134]
[19-Jul-2019 18:48:56] at org.bukkit.craftbukkit.v1_14_R1.event.CraftEventFactory.doEntityAddEventCalling(CraftEventFactory.java:534) ~[patched_1.14.3.jar:git-Paper-134]
[19-Jul-2019 18:48:56] at net.minecraft.server.v1_14_R1.WorldServer.addEntity0(WorldServer.java:971) ~[patched_1.14.3.jar:git-Paper-134]
[19-Jul-2019 18:48:56] at net.minecraft.server.v1_14_R1.WorldServer.addEntity(WorldServer.java:893) ~[patched_1.14.3.jar:git-Paper-134]
[19-Jul-2019 18:48:56] at net.minecraft.server.v1_14_R1.SpawnerCreature.spawnMobs(SpawnerCreature.java:121) ~[patched_1.14.3.jar:git-Paper-134]
[19-Jul-2019 18:48:56] at net.minecraft.server.v1_14_R1.ChunkProviderServer.lambda$tickChunks$7(ChunkProviderServer.java:451) ~[patched_1.14.3.jar:git-Paper-134]
[19-Jul-2019 18:48:56] at it.unimi.dsi.fastutil.longs.Long2ObjectLinkedOpenHashMap$1.forEach(Long2ObjectLinkedOpenHashMap.java:1661) ~[patched_1.14.3.jar:git-Paper-134]
[19-Jul-2019 18:48:56] at com.google.common.collect.Iterables$UnmodifiableIterable.forEach(Iterables.java:105) ~[patched_1.14.3.jar:git-Paper-134]
[19-Jul-2019 18:48:56] at net.minecraft.server.v1_14_R1.ChunkProviderServer.tickChunks(ChunkProviderServer.java:400) ~[patched_1.14.3.jar:git-Paper-134]
[19-Jul-2019 18:48:56] at net.minecraft.server.v1_14_R1.ChunkProviderServer.tick(ChunkProviderServer.java:370) ~[patched_1.14.3.jar:git-Paper-134]
[19-Jul-2019 18:48:56] at net.minecraft.server.v1_14_R1.WorldServer.doTick(WorldServer.java:264) ~[patched_1.14.3.jar:git-Paper-134]

[19-Jul-2019 18:48:56] at net.minecraft.server.v1_14_R1.MinecraftServer.b(MinecraftServer.java:1192) ~[patched_1.14.3.jar:git-Paper-134]
[19-Jul-2019 18:48:56] at net.minecraft.server.v1_14_R1.DedicatedServer.b(DedicatedServer.java:420) ~[patched_1.14.3.jar:git-Paper-134]
[19-Jul-2019 18:48:56] at net.minecraft.server.v1_14_R1.MinecraftServer.a(MinecraftServer.java:1064) ~[patched_1.14.3.jar:git-Paper-134]
[19-Jul-2019 18:48:56] at net.minecraft.server.v1_14_R1.MinecraftServer.run(MinecraftServer.java:908) ~[patched_1.14.3.jar:git-Paper-134]
[19-Jul-2019 18:48:56] at java.lang.Thread.run(Thread.java:748) [?:1.8.0_131]

**Comment by kickash32** [comment]
`[DMS] Error: unknown mob: PHANTOM` will be fixed in the next build (eta next week)
@robertoccu you seem to be having a separate issue, can you open a new issue and include a full server log?

**Comment by kickash32** [comment]
This issue is fixed with build v2.2.0
